src/RestControllers/FHIR/FhirMetaDataRestController.php

```diff
@@ -169,6 +204,85 @@ protected function buildCapabilityStatement()
+        $coding->setCode("SMART-on-FHIR");
+
+        $service->addCoding($coding)
+                ->setText(xlt("OAuth2 using SMART-on-FHIR profile (see http://docs.smarthealthit.org)"));
+
+        $security = new FHIRCapabilityStatementSecurity();
+        $security->addService($service);
+        $this->addOauthSecurityExtensions($security);
+        return $security;
+    }
+
+    /**
+     * Adds all of the FHIR REST Extensions needed for things such as SMART on FHIR
+     * @param FHIRCapabilityStatementSecurity $statement
+     */
+    private function addOauthSecurityExtensions(FHIRCapabilityStatementSecurity $statement)
+    {
+        $authServer = new AuthorizationController();
+        $oauthExtension = new FHIRExtension();
+        $oauthExtension->setUrl(new FHIRUrl("http://fhir-registry.smarthealthit.org/StructureDefinition/oauth-uris"));
+        $oauthUrls = [
+            // @see http://www.hl7.org/fhir/smart-app-launch/StructureDefinition-oauth-uris.html
+            // and @see http://www.hl7.org/fhir/smart-app-launch/conformance/index.html#declaring-support-for-oauth2-endpoints
+            // token and authorize are required because we don't use implicit grant flow.
+            'token' => $authServer->getTokenUrl()
+            ,'authorize' => $authServer->getAuthorizeUrl()
+            ,'register' => $authServer->getRegistrationUrl()
+            ,'introspect' => $authServer->getIntrospectionUrl()
+            // TODO: if we have these URIs we can provide them
+//            ,'manage' => $authServer->getManageUrl()
+//            ,'revoke' => ''
+        ];
+        foreach ($oauthUrls as $url => $valueUri) {
+            $oauthEndpointExtension = new FHIRExtension();
+            $oauthEndpointExtension->setUrl($url);
+            $oauthEndpointExtension->setValueUri($valueUri);
+            $oauthExtension->addExtension($oauthEndpointExtension);
+        }
+        $statement->addExtension($oauthExtension);
+
+        // now add our SMART capabilities
+        foreach (self::SMART_CAPABILITIES as $smartCapability) {
+            $extension = new FHIRExtension();
+            $extension->setUrl("http://fhir-registry.smarthealthit.org/StructureDefinition/capabilities");
+            $extension->setValueCode($smartCapability);
+            $statement->addExtension($extension);
+        }
+    }
+
+    /**
+     * Given an endpoint returns the base URL for it.
+     * @param $endpoint
+     * @return FHIRUrl
+     */
+    private function getFHIRUrl($endpoint)
+    {
+        $gbl = \RestConfig::GetInstance();
+        // TODO: shouldn't this be an HTTPS url?  also this originally was SERVER_NAME but that doesn't include
+        // the port, don't we want the port information?
+        $hostname = $_SERVER['SERVER_NAME'];
+        if (!empty($_SERVER['SERVER_PORT']) && $_SERVER['SERVER_PORT'] != 443 && $_SERVER['SERVER_PORT'] != 80) {
```

Comment:
@sjpadgett Here is where I'm using the SERVER_PORT and SERVER_NAME.  What should I be using here instead?

Reply:
this is set in Globals and is required $GLOBALS['site_addr_oath']
![image](https://user-images.githubusercontent.com/1859985/101967811-30ad9380-3bea-11eb-92aa-04f0368e5b8c.png)

Reply:
to be clear this global should contain port if used also.